src/logger.cc

```diff
@@ -0,0 +1,22 @@
+#include "public/logger.h"
```

Comment:
Obviously nothing to worry about this early in the patch series, but something to consider for future improvements would be support for alternative logging systems.

Theres a lot of ways it can be approached. Macros are an "easy" approach, where ninja defines a macro collection that gets used in specific well defined ways, and library users can define their own.

Template arguments passed in to replace the logging functions at compile time.

Lots of options.